test/retained_topic_map.h

```diff
@@ -0,0 +1,199 @@
+//
```

Comment:
What's substantially different between the retained topic map, and the regular subscription map?

Reply:
I added extra comments to explain